### PR TITLE
Introduce removeNonEmptyDirectory attribute for file systems

### PR DESCRIPTION
**Pull request by alisihab**

### CODE REVIEW

File: core/src/main/java/nl/nn/adapterframework/filesystem/FileSystemActor.java [hunk 1]
```diff
@@ -691,4 +690,12 @@ public void setExcludeWildCard(String excludeWildCard) {
 	public String getExcludeWildCard() {
 		return excludeWildCard;
 	}
+
+	@IbisDoc({"12", "If set to true then the directory and the content of the non empty directory will be deleted."})
+	public void setRemoveNonEmptyDirectory(boolean removeNonEmptyDirectory) {
```
Comment (gvanbrakel) [inline core/src/main/java/nl/nn/adapterframework/filesystem/FileSystemActor.java:695]: Please use 'Folder' instead of 'Directory'

File: core/src/test/java/nl/nn/adapterframework/filesystem/FileSystemActorTest.java [hunk 2]
```diff
@@ -1167,7 +1167,39 @@ public void fileSystemActorRmdirActionTest() throws Exception {
 		// test
 		assertFalse("Expected folder [" + folder + "] " + "not to be present", actual);
 	}
+	@Test
+	public void fileSystemActorRmNonEmptyDirActionTest() throws Exception {
```
Comment (gvanbrakel) [inline core/src/test/java/nl/nn/adapterframework/filesystem/FileSystemActorTest.java:1171]: You should also test that if removeNonEmptyDirecty=false, that an exception is thrown if the folder is not empty.

**Review by gvanbrakel — CHANGES_REQUESTED** [review]

**Review by gvanbrakel — APPROVED** [review]

**Review by nielsm5 — APPROVED** [review]
